sv_inline.h

```diff
@@ -1003,6 +1003,90 @@ Perl_sv_setpv_freshbuf(pTHX_ SV *const sv)
     return SvPVX(sv);
 }
 
+/* Perl_int2str_table: lookup table containing string representations
+ * of all two digit numbers. For example, int2str_table.arr[0] is "00"
+ * and int2str_table.arr[12*2] is "12".
+ *
+ * We are going to read two bytes at a time, so we have to ensure that
+ * the array is aligned to a 2 byte boundary. That's why it was made a
+ * union with a dummy U16 member. */
+static const union {
+    char arr[200];
+    U16 dummy;
+} Perl_int2str_table = {{
+    '0', '0', '0', '1', '0', '2', '0', '3', '0', '4', '0', '5', '0', '6',
+    '0', '7', '0', '8', '0', '9', '1', '0', '1', '1', '1', '2', '1', '3',
+    '1', '4', '1', '5', '1', '6', '1', '7', '1', '8', '1', '9', '2', '0',
+    '2', '1', '2', '2', '2', '3', '2', '4', '2', '5', '2', '6', '2', '7',
+    '2', '8', '2', '9', '3', '0', '3', '1', '3', '2', '3', '3', '3', '4',
+    '3', '5', '3', '6', '3', '7', '3', '8', '3', '9', '4', '0', '4', '1',
+    '4', '2', '4', '3', '4', '4', '4', '5', '4', '6', '4', '7', '4', '8',
+    '4', '9', '5', '0', '5', '1', '5', '2', '5', '3', '5', '4', '5', '5',
+    '5', '6', '5', '7', '5', '8', '5', '9', '6', '0', '6', '1', '6', '2',
+    '6', '3', '6', '4', '6', '5', '6', '6', '6', '7', '6', '8', '6', '9',
+    '7', '0', '7', '1', '7', '2', '7', '3', '7', '4', '7', '5', '7', '6',
+    '7', '7', '7', '8', '7', '9', '8', '0', '8', '1', '8', '2', '8', '3',
+    '8', '4', '8', '5', '8', '6', '8', '7', '8', '8', '8', '9', '9', '0',
+    '9', '1', '9', '2', '9', '3', '9', '4', '9', '5', '9', '6', '9', '7',
+    '9', '8', '9', '9'
+}};
+
+/* uiv_2buf() was originally a private routine in sv.c for use by
+ * sv_2pv_flags(), but its usefulness elsewhere was noted, and it was
+ * moved out here. It prints an IV or UV as a string towards the end
+ * of buf, and return pointers to start and end of it.
+ *
+ * We assume that buf is at least TYPE_CHARS(UV) long.
+ */
+
+PERL_STATIC_INLINE char *
+S_uiv_2buf(char *const buf, const IV iv, UV uv, const int is_uv, char **const peob)
+{
+    char *ptr = buf + TYPE_CHARS(UV);
+    char * const ebuf = ptr;
+    U16 *word_ptr, *word_table;
+
+    PERL_ARGS_ASSERT_UIV_2BUF;
+
+    /* ptr has to be properly aligned, because we will cast it to U16* */
+    assert(PTR2nat(ptr) % 2 == 0);
+    /* we are going to read/write two bytes at a time */
+    word_ptr = (U16*)ptr;
+    word_table = (U16*)Perl_int2str_table.arr;
```

Comment:
This was a problem in the original, but there word_table could be a const pointer which would eliminate the need to cast away the const here (though not the cast entirely)